### ISSUE TITLE
This plugin is ignoring the type when using the "keyvalue" pipe.

### ISSUE THREAD

**Issue by infacto** [angular/vscode-ng-language-service#946]
**Describe the bug**

Using the `keyvalue` pipe throws errors in template.

```text
Identifier 'myProperty' is not defined. 'V' does not contain such a member. (ng)
```

Because this error is only in the VS Code html file, I guess that this is related to this [extension](https://marketplace.visualstudio.com/items?itemName=Angular.ng-template). The build works.

**To Reproduce**

1. Use the `keyvalue` pipe on a property with a custom type. See my source code blocks / graphics or download the demo (see below).
2. See that all properties of `value` are marked as error. The pipe does not pass the type correctly.
3. Build it. It works. But the error in editor persists. It seems to be related to the ng extension. 
4. Try restart of VS Code. Nothing helped.

**Source Code**

app.component.html
```html
<ul>
  <li *ngFor="let entry of menuService.menu | keyvalue">
    <span>{{ entry.value.text }}</span>
  </li>
</ul>
```

menu-items.model.ts
```ts
export interface MenuItem {
  text: string;
}

export interface Menu {
  [key: string]: MenuItem;
}
```

Other (Snippets):
```ts
// app.component.ts
import { MenuService } from './menu-items/menu-items.service';
export class AppComponent {
  constructor(public menuService: MenuService) {}
}

// menu-items.service.ts
import { Menu, MenuItem } from './menu-items.model';
public menu: Menu = {};
// See model (above).
```

**Expected behavior**

No error when using the `keyvalue` pipe. I think it's a correct use of it. It works everywhere.

**Logs**

```text
Identifier 'text' is not defined. 'V' does not contain such a member. (ng)
```

**Screenshots**

![demo](https://user-images.githubusercontent.com/60390085/97168146-3e1cd480-1788-11eb-929d-ea4b82aa7f9a.gif)
(Gif Animation. Wait for the first frame in `app.component.html`)

**Additional context**

[**Download** Demo Project](https://github.com/angular/vscode-ng-language-service/files/5438507/keyvalue-pipe-test.zip)

**Versions**

VS Code:

```text
Version: 1.50.1 (user setup)
Commit: d2e414d9e4239a252d1ab117bd7067f125afd80a
Date: 2020-10-13T15:06:15.712Z
Electron: 9.2.1
Chrome: 83.0.4103.122
Node.js: 12.14.1
V8: 8.3.110.13-electron.0
OS: Windows_NT x64 10.0.19041
```

Angular Language Service (VS Code Extension): `0.1001.0`

Angular CLI (`ng --version`):

```text
Angular CLI: 11.0.0-rc.0
Node: 10.22.1
OS: win32 x64

Angular: 11.0.0-rc.0    
... animations, cli, common, compiler, compiler-cli, core, forms
... language-service, platform-browser, platform-browser-dynamic
... router
Ivy Workspace: Yes

Package                         Version
---------------------------------------------------------
@angular-devkit/architect       0.1100.0-rc.0
@angular-devkit/build-angular   0.1100.0-rc.0
@angular-devkit/core            11.0.0-rc.0
@angular-devkit/schematics      11.0.0-rc.0
@schematics/angular             11.0.0-rc.0
@schematics/update              0.1100.0-rc.0
rxjs                            6.6.3
typescript                      4.0.3
```

**Info**

Created a new Angular app for this demo using the Angular CLI 11 with default settings. No special configs. Just a simple single app repository.

See current [Angular repo pipe source code](https://github.com/angular/angular/blob/master/packages/common/src/pipes/keyvalue_pipe.ts).

**Workaround**

No solution yet. It's really annoying. We need a solution. Any guesses?

**Comment by y0nd0** [comment]
Same here. Tested with the latest version.

![image](https://user-images.githubusercontent.com/68586847/97789679-a0c80300-1bc2-11eb-8488-ac71ba58ba22.png)

**Comment by infacto** [comment]
The output log of Angular Language Service is mostly empty. But on mouse interaction there are some errors about `No config file for`. e.g. 

```
[Error - 10:57:44] No config file for repo-root\node_modules\@angular\common\common.d.ts
```

**Comment by infacto** [comment]
@kyliau Hi, could you please help me? I have no idea, but this repo looks really dead.

**Comment by infacto** [comment]
Ok, I found out that the pipe from Anglar `11.0.0-next.3` works but `11.0.0-next.4` does not work. Moving to Angular common.

**Comment by infacto** [comment]
Ok, in the referenced issue you can see.

> This is a bug in the language service.

**Comment by ayazhafiz** [comment]
As @infacto mentioned, this is a known issue (angular/angular#21224) and is fixed by the Ivy language service, which will be beta released in Angular 11 (#335 (comment)). As such, I am closing this report.
